kernel/packages/shared/session/sagas.ts

```diff
@@ -168,7 +169,7 @@ function* startSignUp(userId: string, identity: ExplorerIdentity) {
   let prevGuest = fetchProfileLocally(userId)
   let profile: Profile = prevGuest ? prevGuest : yield generateRandomUserProfile(userId)
   profile.userId = identity.address.toString()
-  profile.ethAddress = identity.address.toString()
+  profile.ethAddress = identity.rawAddress ? identity.rawAddress.toString() : profile.userId
```

Comment:
Why do you need to call `toString`?

Reply:
Also, according to the types you defined below this could never be undefined, so this check seems unnecessary

Reply:
Done